### PR TITLE
Endpoint to include more data with AST node metrics

### PR DESCRIPTION
**Pull request by intjftw**
As it was requested, I created a new endpoint that includes more AST node data along with the list of available metrics. It is in WIP status as I still need to add more data and test the endpoint.

### CODE REVIEW

File: plugins/cpp_metrics/service/cxxmetrics.thrift [hunk 1]
```diff
@@ -90,6 +103,15 @@ service CppMetricsService
   map<common.AstNodeId, list<CppMetricsAstNodeSingle>> getCppAstNodeMetricsForPath(
     1:string path)
 
+  /**
+   * This function returns all available C++ metrics
+   * (AST node-level) and miscellaneous data for a particular path.
+   *
+   * The given path is a handled as a prefix.
+   */
+  map<common.AstNodeId, CppMetricsAstNodeAllData> getCppAstNodeMetricsAndDataForPath(
```
Comment (mcserep) [inline plugins/cpp_metrics/service/cxxmetrics.thrift:112]: Suggested name: `getCppAstNodeMetricsDetailedForPath`

File: plugins/cpp_metrics/service/cxxmetrics.thrift [hunk 2]
```diff
@@ -38,10 +38,16 @@ struct CppMetricsAstNodeSingle
   3:double value
 }
 
-struct CppMetricsAstNodeAll
```
Comment (mcserep) [inline plugins/cpp_metrics/service/cxxmetrics.thrift:41]: Was this struct unused previously, that you can change it?

File: plugins/cpp_metrics/service/cxxmetrics.thrift [hunk 3]
```diff
@@ -38,10 +38,16 @@ struct CppMetricsAstNodeSingle
...
+struct CppMetricsAstNodeAllData
```
Comment (mcserep) [inline plugins/cpp_metrics/service/cxxmetrics.thrift:41]: Well, it's not all data. Suggested name: `CppMetricsAstNodeDetailed`.

File: plugins/cpp_metrics/service/cxxmetrics.thrift [hunk 4]
```diff
@@ -74,6 +80,13 @@ service CppMetricsService
   list<CppMetricsAstNodeSingle> getCppMetricsForAstNode(
     1:common.AstNodeId astNodeId)
 
+  /**
+   * This function returns all available C++ metrics
+   * for a particular AST node.
+   */
+  //list<CppMetricsAstNodeSingle> getCppMetricsAndDataForAstNode(
+    //1:common.AstNodeId astNodeId)
```
Comment (mcserep) [inline plugins/cpp_metrics/service/cxxmetrics.thrift:88]: Do not add commented out code to the codebase.

**Review by mcserep — CHANGES_REQUESTED** [review]
Seems fine apart from minor issues.

File: plugins/cpp_metrics/service/cxxmetrics.thrift (hunk 2 above)
Comment (intjftw) [reply plugins/cpp_metrics/service/cxxmetrics.thrift:41]: Yes, I checked, and it wasn't used anywhere.

**Review by mcserep — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**